x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts

```diff
@@ -29,12 +29,12 @@ const mockExpectedResponseHeaders = {
   [READ_ONLY_MODE_HEADER]: 'false',
 };
 
-describe('EnterpriseSearchRequestHandler', () => {
-  const enterpriseSearchRequestHandler = new EnterpriseSearchRequestHandler({
-    config: mockConfig,
-    log: mockLogger,
-  }) as any;
+const enterpriseSearchRequestHandler = new EnterpriseSearchRequestHandler({
+  config: mockConfig,
+  log: mockLogger,
+}) as any;
 
+describe('EnterpriseSearchRequestHandler', () => {
```

Comment:
[nit] Can we revert this change and where the const was moved to? Across our Kibana codebase, we've generally been putting constants/mock data used by specific describe blocks within that block.

Reply:
I need access to that constant below and (based on my understanding of Javascript scoping, which may not be correct), `enterpriseSearchRequestHandler.httpAgent` requires the const to be at the top level. I may be wrong there, going to check now...

Reply:
```
    ReferenceError: enterpriseSearchRequestHandler is not defined

      477 |       method: 'GET',
      478 |       body: undefined,
    > 479 |       agent: enterpriseSearchRequestHandler.httpAgent,
          |              ^
      480 |       ...expectedParams,
      481 |     });
      482 |   },
      ```

Reply:
Ah I see, apologies, I should have opened the full file instead of skimming the git diff.

So the best approach here instead is to not actually rely on `enterpriseSearchRequestHandler` for a http.Agent - instead, what we should do is simply mock it:

```js
// line 24 or thereabouts
const mockHttpAgent = jest.fn();
jest.mock('./enterprise_search_http_agent', () => ({
  entSearchHttpAgent: mockHttpAgent,
}));

// line 479
agent: mockHttpAgent,
```

Let me know if that makes sense - this is a generally a strong perf advantage as well as following the principles of unit testing - IMO, the reason why we split things out into separate files is so other files can easily mock it and not worry about what it actually does

Reply:
```
    ReferenceError: Cannot access 'mockHttpAgent' before initialization

      32 | const mockHttpAgent = jest.fn();
      33 | jest.mock('./enterprise_search_http_agent', () => ({
    > 34 |   entSearchHttpAgent: mockHttpAgent,
         |                       ^
      35 | }));
```

Reply:
good times with JS fighting Jest hoisting. In that case I'd suggest just a basic mock->import style:

```js
jest.mock('./enterprise_search_http_agent', () => ({
  entSearchHttpAgent: jest.fn(),
}));
import { entSearchHttpAgent } from './enterprise_search_http_agent';

// line 479
agent: entSearchHttpAgent,